kxml_compiler/CMakeLists.txt

```diff
@@ -21,11 +20,14 @@ set(KXML_COMPILER_HEADERS
 	writercreator.h
 )
 
-add_executable(kxml_compiler ${KXML_COMPILER_SOURCES} ${KXML_COMPILER_HEADERS})
+add_library(libkxml_compiler OBJECT ${KXML_COMPILER_SOURCES} ${KXML_COMPILER_HEADERS})
+target_link_libraries(libkxml_compiler Qt5::Xml)
+add_executable(kxml_compiler kxml_compiler.cpp)
 
 target_link_libraries(kxml_compiler
 	Qt5::Xml
 	kode
+	libkxml_compiler
```

Comment:
Interesting. The CMake documentation for "OBJECT" libraries says you should use `$<TARGET_OBJECTS:objlib>` in the `add_executable` call. It doesn't mention linking to the lib so I'm surprised that it works.
Does it create an actual library on disk (it shouldn't), or is cmake more clever than its documentation indicates, and it instead adds the .o files to the link step of the executable?

Reply:
cmake produces pseudo library target *.objlib which is simple empty text file, and links all *.o of that dummy library. I don't see any issues about this.